symengine/tests/basic/test_relationals.cpp

```diff
@@ -228,6 +231,28 @@ TEST_CASE("Boolean Values", "[Relationals]")
     CHECK_THROWS_AS(Le(I, one), SymEngineException);
 }
 
+TEST_CASE("Logical Not", "[Relationals]")
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const Symbol> y = symbol("y");
+
+    RCP<const Boolean> a = Eq(x, zero);
```

Comment:
@isuruf The errors are arising here as `Eq` or even `Ne`, `Le` and `Lt` return `RCP<const Basic>` objects. I can't seem to find a workaround, what do you suggest.